packages/legend-extension-dsl-data-space/src/stores/DataSpaceViewerState.ts

```diff
@@ -0,0 +1,200 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type {
+  ClassView,
+  DiagramRenderer,
+} from '@finos/legend-extension-dsl-diagram';
+import {
+  Diagram,
+  DIAGRAM_INTERACTION_MODE,
+} from '@finos/legend-extension-dsl-diagram';
+import type {
+  GraphManagerState,
+  PackageableElementReference,
+  PackageableRuntime,
+} from '@finos/legend-graph';
+import { guaranteeNonNullable } from '@finos/legend-shared';
+import { action, computed, makeObservable, observable } from 'mobx';
+import type {
+  ResolvedDataSpace,
+  ResolvedDataSpaceExecutionContext,
+} from '../models/protocols/pure/DSLDataSpace_PureProtocolProcessorPlugin';
+
+export enum DATA_SPACE_VIEWER_ACTIVITY_MODE {
+  MODELS_OVERVIEW = 'MODELS_OVERVIEW',
+  EXECUTION = 'EXECUTION',
+  ENTITLEMENT = 'ENTITLEMENT',
+  TEST_DATA = 'TEST_DATA',
+  SUPPORT = 'SUPPORT',
+}
+
+export class DataSpaceViewerState {
+  graphManagerState: GraphManagerState;
+  dataSpaceGroupId: string;
```

Comment:
if we already have the `ResolvedDataSpace` why do we need the groupId, artifactId, and versionId as separate properties in this state?